conn.go

```diff
@@ -495,6 +518,16 @@ func (conn *Conn) sendReply(dest string, serial uint32, values ...interface{}) {
 	conn.outLck.RUnlock()
 }
 
+func (conn *Conn) defaultSignalAction(fn func(h *defaultSignalHandler, ch chan<- *Signal), ch chan<- *Signal) {
+	conn.signalLck.RLock()
+	defer conn.signalLck.RUnlock()
+	if !isDefaultSignalHandler(conn.signalHandler) {
```

Comment:
Any handler that didn't want to implement the interface could do what you have done here, and simply return. The main advantage is just keeping the code here simple.